internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -510,10 +516,23 @@ func (r *ClusterDeploymentsReconciler) createNewDay2Cluster(
 	spec := clusterDeployment.Spec
 	id := strfmt.UUID(uuid.New().String())
 	apiVipDnsname := fmt.Sprintf("api.%s.%s", spec.ClusterName, spec.BaseDomain)
+
+	pullSecret, err := getPullSecret(ctx, r.Client, spec.PullSecretRef.Name, key.Namespace)
+	if err != nil {
+		r.Log.WithError(err).Error("failed to get pull secret")
```

Comment:
Same as above, if not found then update conditions and let the user know that the link is broken.
Otherwise requeue.

Reply:
sure, used 'r.updateStatus' instead as done for other error handling